packages/app/src/vis-packs/nexus/pack-utils.ts

```diff
@@ -2,57 +2,49 @@ import {
   assertGroupWithChildren,
   assertStr,
   buildEntityPath,
-  handleError,
-  hasChildren,
   hasComplexType,
   hasMinDims,
   hasNumDims,
   isGroup,
 } from '@h5web/shared';
 import type { Entity } from '@h5web/shared';
-import type { FetchStore } from 'react-suspense-fetch';
 
-import { ProviderError } from '../../providers/models';
 import { getAttributeValue } from '../../utils';
 import type { VisDef } from '../models';
 import { NxInterpretation } from './models';
-import { findSignalDataset, isNxDataGroup, isNxGroup } from './utils';
+import { findSignalDataset, hasNxClass, isNxDataGroup } from './utils';
 import { NexusVis, NEXUS_VIS } from './visualizations';
 
-function getDefaultEntityPath(entity: Entity): string | undefined {
-  const defaultPath = getAttributeValue(entity, 'default');
-  if (defaultPath) {
-    assertStr(defaultPath, `Expected 'default' attribute to be a string`);
-    return defaultPath;
-  }
+function getImplicitDefaultChild(children: Entity[]): Entity | undefined {
+  const groups = children.filter(isGroup);
 
-  if (!isGroup(entity) || !hasChildren(entity)) {
-    return undefined;
+  // Look for an `NXdata` child group first
+  const nxDataChild = groups.find((g) => hasNxClass(g, 'NXdata'));
+  if (nxDataChild) {
+    return nxDataChild;
   }
 
-  const nxChild = entity.children.find((c) => isNxGroup(c));
-  return nxChild?.path;
+  // Then for an `NXentry` child group
+  return groups.find((g) => hasNxClass(g, 'NXentry'));
```

Comment:
I wonder... If this checks fails, should we have a fallback to `NXprocess` as well, given the similarity with `NXentry` ?

Reply:
Yeah not sure. Maybe you're right. That would actually be more coherent with the NX badge logic.

Reply:
done